### ISSUE TITLE
Improve validation of children

### ISSUE THREAD

**Issue by denyo** [porsche-design-system/porsche-design-system#1972]
Scope
---
With #1842 we made a big step towards unified validation within all components. While primarily focused on the validation of properties, also parent and child validation was aligned and improved. 

However, the current child validation happens only once via `componentWillLoad()` if if happens at all.  
This is better than no validation but any later added or removed child might break the component. 

To solve this we should observe the removal and adding of children and validate again.

Out of Scope
---
List topics that are not part of this task.

Notes  
---
- Rework all `this.host.children` like stuff (e.g. in `p-tabs` and `p-stepper-horizontal`)
- consider reusing `throwIfChildrenAreNotOfKind()` util
- `ComponentsMeta` contains a `hasSlot: boolean` for each component which may be useful
- validation should be as precise as possible (e.g. first level, only one child, etc. to not break `::slotted` selectors) 

Acceptance criteria
---
- [ ] Criteria

Subtasks
---
- [ ] Task
